src/extension/handler/uri.ts

```diff
@@ -35,24 +36,31 @@ import {
 const REGEX_WEB_RESOURCE = /^https?:\/\//
 const log = getLogger('RunmeUriHandler')
 
-export class RunmeUriHandler implements UriHandler {
+export class RunmeUriHandler extends EventEmitter<Uri> implements UriHandler {
```

Comment:
I'm wondering if composition vs. inheritance is a better pattern here? Mainly because the UriHandler handles different "commands". It'd be different if we could register a UriHandler per command (which seems excessive).

Check out `runmeServer.ts` for an example how to use Event/EventEmitter as properties.

Reply:
It was a first try, probably there is a better way to do it, the only thing that I'm pretty sure of is that can only register a single URI handler in its entire activation lifetime. https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L10961

Let me review the **runmeServer.ts**.